web/packages/quick-editor/src/quick-editor-core.ts

```diff
@@ -0,0 +1,94 @@
+const ScopeList = [
+	'about',
+	'avatars',
+	'verified-accounts',
+	'links',
+	'interests',
+	'contact-info',
+	'wallet',
+	'photos',
+	'design',
+	'privacy'
+] as const;
+
+export type Scope = typeof ScopeList[number][];
+
+export type ProfileUpdatedType = 'avatar_updated';
+
+export type Open = ( email?: string ) => void;
+
+export type OnProfileUpdated = ( type: ProfileUpdatedType ) => void;
+
+export type OnOpened = () => void;
+
+export type QuickEditorCoreOptions = Partial<{
+	email: string;
+	scope: Scope;
+	local: string;
+	onProfileUpdated: OnProfileUpdated;
+	onOpened: OnOpened;
+}>;
+
+export class GravatarQuickEditorCore {
+	_name: string;
+	_email: string;
+	_scope: Scope;
+	_local: string;
+	_onProfileUpdated: OnProfileUpdated;
+	_onOpened: OnOpened;
+
+	constructor( { email, scope = [], local, onProfileUpdated, onOpened }: QuickEditorCoreOptions ) {
+		this._name = this._getName();
+		this._email = email;
+		this._scope = scope;
+		this._local = local;
+		this._onProfileUpdated = onProfileUpdated;
+		this._onOpened = onOpened;
+
+		if ( ! this._scope.every( s => ScopeList.includes( s ) ) ) {
+			console.error( 'Gravatar Quick Editor: Invalid scope definition. Available scope: ' + ScopeList.join( ', ' ) );
+			this._scope = this._scope.filter( s => ScopeList.includes( s ) );
+		}
+
+		window.addEventListener( 'message', this._onMessage.bind( this ) );
+	}
+
+	open: Open = ( email?: string ) => {
```

Comment:
Since we have already assigned the type to the method, we can remove the duplicated arg type:

```suggestion
	open: Open = ( email ) => {
```

Reply:
✅